spec/ics-2-consensus-verification/README.md

```diff
@@ -0,0 +1,265 @@
+is possible that the light clients' behaviour differs from the full nodes'. In this case, the
+equivocation proof which proves the divergence between the `Verifier` and the full node will be
+generated and submitted to the chain, as defined in
+[ICS ?](https://github.com/cosmos/ics/issues/53), so the chain can safely deactivate the
+light client.
+
+### Technical Specification
+
+#### Requirements
+
+* `get`, `set`, `Key`, `Identifier`, as defined in [ICS24](https://github.com/cosmos/ics/pull/75),
+are used by the datagram handler.
+
+* `CommitmentRoot`, as defined in [ICS23](https://github.com/cosmos/ics/pull/74), is used by
+`ConsensusState`. The downstream logic can use it to verify whether key-value pairs are present
+in the state or not.
+
+#### Definitions
+
+##### ConsensusState
+
+`ConsensusState` is a blockchain commit which contains an `CommitmentRoot` and the requisite
+state to verify future roots, stored in one blockchain to verify the state of the other.
+Defined as
+
+```go
+type ConsensusState struct {
+  Base VerifierBase
+  Root CommitmentRoot
+}
+```
+where
+  * `Base` is a data used by `Consensus.Verifier` to verify `Header`s
+  * `Root` is the `CommitmentRoot`, used to prove internal state
+
+##### Header
+
+`Header` is a blockchain header which provides information to update `ConsensusState`,
+submitted to one blockchain to update the stored `ConsensusState`.
+Defined as
+
+```go
+type Header struct {
+  Proof HeaderProof
+  Base Maybe[VerifierBase]
+  Root CommitmentRoot
+}
+```
+where
+  * `Proof` is the commit proof used by `Consensus.Verifier` to be verified
+  * `Base` is the new verify, if needed to be updated
+  * `Root` is the new `CommitmentRoot` which will replace the existing one
+
+##### Verifier
+
+`Verifier` is a light client verifier proving `Header` depending on the `Commit`.
+It SHOULD prove far more efficiently than replaying `Consensus` logic
+for the given parent `Header` and the list of network messages, ideally in O(1) time.
+Defined as
+
+```go
+type Verifier func(ConsensusState, Header) (Error|ConsensusState)
+```
+
+The detailed specification of `Verifier` is defined in [VERIFIER.md](./VERIFIER.md)
+
+##### LightClient
+
+LightClient is defined as
+```go
+type LightClient struct {
+  Verifier Verifier
+  ConsensusState ConsensusState
+}
+```
+where
+  * `ConsensusState` is the root of trust providing the `VerifierBase`
+  * `Verifier` is the lightclient verification logic
+
+The exact type of each fields are depending on the type of the actual consensus logic.
+
+#### Subprotocols
+
+The chains MUST implement function `register` and `update`, as they form the `handleDatagram`.
+Calling both functions MAY be permissionless.
+
+##### Preliminaries
+
+`newID` is a function which generates a new `Identifier` for a `LightClient`, which MAY depending
+on the `Header`. The behaviour of `newID` is implementation specific. Possible implementations are:
```

Comment:
shouldn't use "the" for `Header`,  as it makes it sound like there is a single `Header` per chain`. We should explicitly specify that it is one header per light client. 

Also whenever we use `may`, we need to specify when it applies or not, or link to something that does specify in which case a header is indeed required

Reply:
still needs to be addressed